libraries/chain/abi_serializer.cpp

```diff
@@ -294,6 +306,13 @@ namespace eosio { namespace chain {
         EOS_ASSERT(_is_type(t.second, ctx), invalid_type_inside_abi, "${type}", ("type",impl::limit_size(t.second)) );
       } FC_CAPTURE_AND_RETHROW( (t)  ) }
 
+      for( const auto& kt : kv_tables ) { try {
+        ctx.check_deadline();
+        EOS_ASSERT(_is_type(kt.second.type, ctx), invalid_type_inside_abi,
+                   "Invalid reference in struct ${type}", ("type", impl::limit_size(kt.second.type)));
+        EOS_ASSERT( kt.second.primary_index.type.size() > 0, invalid_type_inside_abi, "missing primary index$ {p}", ("p",impl::limit_size(kt.first.to_string())));
+      } FC_CAPTURE_AND_RETHROW( (impl::limit_size(kt.first.to_string()))) }
```

Comment:
This does not produce valid json which can be problematic for things that report exceptions in json format.
If `kt.first` is of type `name` then the `limit_size` is not needed. 
I would just not capture anything here.

Reply:
removed try FC_CAPTURE_AND_RETHROW